src/codeu/chat/client/simplegui/UserPanel.java

```diff
@@ -144,27 +151,89 @@ private void initialize() {
       public void actionPerformed(ActionEvent e) {
         UserPanel.this.getAllUsers(listModel);
       }
+    });   
+
+     userSignInButton.addActionListener(new ActionListener() {
+      @Override
+      public void actionPerformed(ActionEvent e) {
+       /* final String s = (String) JOptionPane.showInputDialog(
+            UserPanel.this, "Enter user name:", "Continue", JOptionPane.PLAIN_MESSAGE,
+            null, null, "");
+
+	final String p = (String) JOptionPane.showInputDialog(
+            UserPanel.this, "Enter Password:", "Sign-In", JOptionPane.PLAIN_MESSAGE,
+            null, null, "");
+*/
+	Login login = new Login(clientUser);
+	loginDialog.setVisible(true);
+
+        final String s = loginDialog.getUsername();
+	final String p = loginDialog.getPassword();
+
+	//System.out.println("Input Username is " + s);
+	//System.out.println("Input Password is " + p);
+
+	final User u = clientUser.searchByName(s);
+
+	//System.out.println("Real user is " + u.name);
+	//System.out.println("Real pass is " + u.password);
+
+	if(u != null){
```

Comment:
u should never be null here, right? I think it's a reasonable guard to have, but might be worth noting that something crazy would have to happen (someone else deletes a user mid-login)